src/app/api/api_v1/endpoints/detections.py

```diff
@@ -64,23 +64,40 @@ async def create_detection(
 @router.get("/{detection_id}", status_code=status.HTTP_200_OK, summary="Fetch the information of a specific detection")
 async def get_detection(
     detection_id: int = Path(..., gt=0),
+    cameras: CameraCRUD = Depends(get_camera_crud),
     detections: DetectionCRUD = Depends(get_detection_crud),
     token_payload: TokenPayload = Security(get_jwt, scopes=[UserRole.ADMIN, UserRole.AGENT, UserRole.USER]),
 ) -> Detection:
     telemetry_client.capture(token_payload.sub, event="detections-get", properties={"detection_id": detection_id})
-    return cast(Detection, await detections.get(detection_id, strict=True))
+    detection = cast(Detection, await detections.get(detection_id, strict=True))
+    camera = await cameras.get(detection.camera_id, strict=True)
+    if (
+        camera is not None
+        and token_payload.organization_id != camera.organization_id
+        and UserRole.ADMIN not in token_payload.scopes
+    ):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Access forbidden.")
+    return detection
 
 
 @router.get("/{detection_id}/url", response_model=DetectionUrl, status_code=200)
 async def get_detection_url(
     detection_id: int = Path(..., gt=0),
+    cameras: CameraCRUD = Depends(get_camera_crud),
     detections: DetectionCRUD = Depends(get_detection_crud),
     token_payload: TokenPayload = Security(get_jwt, scopes=[UserRole.ADMIN, UserRole.AGENT, UserRole.USER]),
 ) -> DetectionUrl:
     """Resolve the temporary media image URL"""
-    telemetry_client.capture(token_payload.sub, event="detections-url", properties={"detection_id": detection_id})
     # Check in DB
+    telemetry_client.capture(token_payload.sub, event="detections-url", properties={"detection_id": detection_id})
```

Comment:
I imagine this was a revert but it's better when the comment matches the line right below it